### ISSUE TITLE
Getting an incorrect error about an unsupported TypeScript version

### ISSUE THREAD

**Issue by kaiyoma** [typescript-eslint/typescript-eslint#3162]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
Install these package versions:

```
    "@typescript-eslint/eslint-plugin": "4.17.0",
    "@typescript-eslint/parser": "4.17.0",
    "typescript": "4.1.5"
```

**Expected Result**
No errors about unsupported versions.

**Actual Result**
```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/types
cript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.2.0

YOUR TYPESCRIPT VERSION: 4.2.3

Please only submit bug reports when using the officially supported version.
```

**Additional Info**
I have no idea why this package thinks I'm running 4.2.3, because I'm definitely not. I'm specifying exactly `"4.1.5"` in my package.json and these commands also show only 4.1.5:

```
$ yarn why typescript
yarn why v1.22.5
[1/4] Why do we have the module "typescript"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "typescript@4.1.5"
info Has been hoisted to "typescript"
info Reasons this module exists
   - "workspace-aggregator-27e3acdb-d684-4a1b-85fd-54f41a42027a" depends on it
   - Specified in "devDependencies"
   - Hoisted from "_project_#typescript"
info Disk size without dependencies: "55.78MB"
info Disk size with unique dependencies: "55.78MB"
info Disk size with transitive dependencies: "55.78MB"
info Number of shared dependencies: 0
Done in 3.21s.
```

```
$ yarn tsc --version
yarn run v1.22.5
$ ...\node_modules\.bin\tsc --version
Version 4.1.5
Done in 0.46s.
```

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `4.14.1` and `4.17.0` |
| `TypeScript`                           | `4.1.5` |
| `node`                                 | `14.15.0` |

**Comment by bradzacher** [comment]
Our tooling simply does `require('typescript')` from the directory within which it's installed.
The version that's reported is taken directly from the typescript package it requires:
https://github.com/typescript-eslint/typescript-eslint/blob/d3086d8acc6092964d98747d782417225871cb0b/packages/typescript-estree/src/parser.ts#L34
Which means if it's reporting `4.2.3`, then something has installed `4.2.3` in your `node_modules` folder.

**Comment by armano2** [comment]
@kaiyoma can you check if `@typescript-eslint/parser` and `@typescript-eslint/typescript-estree` is picked correctly, as it seem that you are having issues with hoisting dependencies.

```bash
yarn why @typescript-eslint/typescript-estree
```

**Comment by kaiyoma** [comment]
@bradzacher @armano2 Ah yes, that was the underlying issue. I still had some old yarn links lying around, but after a clean install, all the package versions match up now and the TypeScript errors have gone away. Thanks for your quick responses!
